File: LeetCode/Easy/0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] counter = new int[58];
        int ret = 0;
        
        for (char v : s.toCharArray()){
            ++counter[v - 65];
        }
        
        for (int i = 0; i < counter.length; ++i){
            ret += counter[i] / 2 * 2;
            if (ret % 2 == 0 && counter[i] % 2 == 1) {
                ++ret;
            }
        }
        
        return ret;
    }
}

File: LeetCode/Easy/0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int counter[58]{0};
        int ret = 0;
        for(int i = 0; i < s.size(); ++i) ++counter[s[i] - 65];
        
        for(int i = 0; i < 58; ++i){
            ret += counter[i] / 2 * 2;
            if (ret % 2 == 0 && counter[i] % 2 == 1) ++ret;
        }
        return ret;
    }
};